XamarinCommunityToolkit/Behaviors/EventToCommandBehavior.shared.cs

```diff
@@ -42,85 +44,56 @@ public object CommandParameter
         }
 
         protected override void OnDetachingFrom(View bindable)
         {
-            DeregisterEvent(EventName);
+            UnregisterEvent();
             base.OnDetachingFrom(bindable);
         }
 
-        void RegisterEvent(string name)
-        {
-            if (string.IsNullOrWhiteSpace(name))
-                return;
+        static void OnEventNamePropertyChanged(BindableObject bindable, object oldValue, object newValue)
+            => ((EventToCommandBehavior)bindable).RegisterEvent();
 
-            var eventInfo = AssociatedObject.GetType().GetRuntimeEvent(name);
-
-            if (eventInfo == null)
-                throw new ArgumentException($"{nameof(EventToCommandBehavior)}: Can't register the {EventName} event.");
-
-            var methodInfo = typeof(EventToCommandBehavior).GetTypeInfo().GetDeclaredMethod(nameof(OnEvent));
-            eventHandler = methodInfo.CreateDelegate(eventInfo.EventHandlerType, this);
-            eventInfo.AddEventHandler(AssociatedObject, eventHandler);
-        }
-
-        void DeregisterEvent(string name)
+        void RegisterEvent()
         {
-            if (string.IsNullOrWhiteSpace(name) || eventHandler == null)
-                return;
+            UnregisterEvent();
 
-            var eventInfo = AssociatedObject.GetType().GetRuntimeEvent(name);
+            var eventName = EventName;
+            if (View == null || string.IsNullOrWhiteSpace(eventName))
+                return;
 
-            if (eventInfo == null)
-                throw new ArgumentException($"{nameof(EventToCommandBehavior)}: Can't de-register the {EventName} event.");
+            eventInfo = View.GetType().GetRuntimeEvent(eventName) ??
+                throw new ArgumentException($"{nameof(EventToCommandBehavior)}: Could't resolve the event.", nameof(EventName));
 
-            eventInfo.RemoveEventHandler(AssociatedObject, eventHandler);
-            eventHandler = null;
+            eventInfo.AddEventHandler(View, eventHandler);
         }
 
-        async void OnEvent(object sender, object eventArgs)
+        void UnregisterEvent()
         {
-            if (Command == null)
-                return;
-
-            var resolvedParameter = CommandParameter
-                ?? Converter.Convert(eventArgs, typeof(object), null, null)
-                ?? eventArgs;
-
-            if (Command.CanExecute(resolvedParameter))
-                Command.Execute(resolvedParameter);
-
-            foreach (var bindable in Actions)
-            {
-                bindable.BindingContext = BindingContext;
-                var action = (IAction)bindable;
-                await action.Execute(sender, eventArgs);
-            }
+            eventInfo?.RemoveEventHandler(View, eventHandler);
+            eventInfo = null;
         }
 
-        static void OnEventNameChanged(BindableObject bindable, object oldValue, object newValue)
+        void OnEventOccurred(object sender = null, object eventArgs = null)
```

Comment:
I think in Forms the "Trigger<event>" is used mostly. Would it make sense to be more consistent and follow that? This is used to trigger the actual Command, so I think that makes a bit more sense

Reply:
renamed